src/Microsoft.Extensions.Logging.Abstractions/Internal/LogValuesFormatter.cs

```diff
@@ -18,24 +18,36 @@ public class LogValuesFormatter
         private const string NullValue = "(null)";
         private static readonly object[] EmptyArray = new object[0];
         private static readonly char[] FormatDelimiters = {',', ':'};
-        private readonly string _format;
+        private readonly Lazy<string> _format;
         private readonly List<string> _valueNames = new List<string>();
 
         public LogValuesFormatter(string format)
         {
             OriginalFormat = format;
+            _format = new Lazy<string>(() => FormatInput(format), true);
+        }
+
+        public string OriginalFormat { get; private set; }
+        public List<string> ValueNames => _valueNames;
```

Comment:
Why?

Reply:
refactored and checked-in. Please review the latest check-in and share review comments.